### PR TITLE
MEN-4654: Handle vexpress board removal in U-Boot

### PR DESCRIPTION
**Pull request by kacf**

### CODE REVIEW

**Comment by kacf** [comment]
@mender-test-bot start pipeline

**Comment by kacf** [comment]
@mender-test-bot start pipeline

**Comment by kacf** [comment]
@mender-test-bot start pipeline

**Comment by kacf** [comment]
@mender-test-bot start pipeline

**Comment by kacf** [comment]
@mender-test-bot start pipeline

**Comment by kacf** [comment]
@lluiscampos: Probably you can skip reviewing the patches for U-Boot in detail, since I plan to submit these upstream and get a review there.

**Comment by kacf** [comment]
@mender-test-bot start pipeline

**Comment by kacf** [comment]
@mender-test-bot start pipeline

File: meta-mender-qemu/recipes-bsp/u-boot/u-boot-vexpress-qemu.inc [hunk 1]
```diff
@@ -1,5 +1,17 @@
 FILESEXTRAPATHS_prepend := "${THISDIR}/patches:${THISDIR}/files:"
 
+# Remove this block when moving from dunfell to the next branch, and also remove
+# the bundled version of u-boot_2021.07. See MEN-3915.
```
Comment (lluiscampos) [inline meta-mender-qemu/recipes-bsp/u-boot/u-boot-vexpress-qemu.inc:4]: Please make a note for this in the task itself too.

**Review by lluiscampos — APPROVED** [review]
The build parts, and the reasoning behind the changes, look good to me. The details on the U-boot patches of course go over my head.

I am very impressed @kacf :medal_sports:

File: meta-mender-qemu/recipes-bsp/u-boot/u-boot-vexpress-qemu.inc (hunk 1 above)
Comment (kacf) [reply meta-mender-qemu/recipes-bsp/u-boot/u-boot-vexpress-qemu.inc:4]: I will. I just want to get this committed first so I can refer to the commit IDs.

**Comment by lluiscampos** [comment]
:tada:

File: meta-mender-qemu/recipes-bsp/u-boot/u-boot-vexpress-qemu.inc (hunk 1 above)
Comment (kacf) [reply meta-mender-qemu/recipes-bsp/u-boot/u-boot-vexpress-qemu.inc:4]: On second thought, I will change this comment. The revert of the bundled version is now tracked implicitly by being included as a revert in [the master-next branch](https://github.com/mendersoftware/meta-mender/commit/cd81dfa80e1d6dbb80a0d02c8b7af7b5471af85a). The removal of the patches listed here doesn't actually depend on which branch we're on, but on whether upstream accepts our patches. I will create a task for tracking the upstream work, and then link to that from here.

Comment (kacf) [reply meta-mender-qemu/recipes-bsp/u-boot/u-boot-vexpress-qemu.inc:4]: https://github.com/mendersoftware/meta-mender/pull/1460 and https://tracker.mender.io/browse/MEN-5023

**Pull request merged**